Sources/Turf/Spline.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+#if !os(Linux)
+import CoreLocation
+#endif
+
+
+struct SplinePoint {
+    let x: CLLocationDegrees
+    let y: CLLocationDegrees
+    let z: CLLocationDegrees
+    
+    init(coordinate: CLLocationCoordinate2D) {
+        x = coordinate.longitude
+        y = coordinate.latitude
+        z = 0
+    }
+    
+    init(x: CLLocationDegrees, y: CLLocationDegrees, z: CLLocationDegrees) {
+        self.x = x
+        self.y = y
+        self.z = z
+    }
+    
+    var coordinate: CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: y, longitude: x)
+    }
+}
+
+struct Spline {
+    private let points: [SplinePoint]
+    private let duration: Int
+    private let sharpness: Double
+    private let stepLength: Double
+    private let length: Int
+    private let delay: Int = 0
+    private var centers = [SplinePoint]()
+    private var controls = [(SplinePoint, SplinePoint)]()
+    private var steps = [Int]()
+    
+    init?(points: [SplinePoint], duration: Int, sharpness: Double, stepLength: Double = 60) {
+        guard points.count >= 2 else { return nil }
+        self.points = points
+        self.duration = duration
+        self.sharpness = sharpness
+        self.stepLength = stepLength
+        
+        length = points.count
+        
+        for index in stride(from: 0, to: points.count - 1, by: 1) {
```

Comment:
A stride of 1 is equivalent to a range:

```swift
center = (0..<points.count).map {
    let points = points[$0]
    …
    return center
}
```

or:

```swift
center = points.enumerated().map {
    let point = $0.1
    let nextPoint = points[$0.0 + 1]
    …
    return center
}
```

A more concise, though possibly less memory-efficient way of writing this loop would be to use `zip(_:_:)`:

```swift
zip(points, points.suffix(from: 1)).map {
    return SplinePoint(x: ($0.x + $1.x) / 2,
                       y: ($0.y + $1.y) / 2,
                       z: ($0.z + $1.z) / 2)
}
```

Reply:
I updated with a range, zip could be an issue for perf on huge lines.

Reply:
Yeah, I think you made the right call. 👍